Save command history to file

NQNStudios opened the pull request:
This provides the feature requested in #17. Some implementation notes:
- I tested this by hand on Arch Linux. The file APIs I used should also work on Mac and Windows, but I can only personally test Linux right now.
- Unit testing this would be squirrelly but possible. Let me know if I should add tests.
- "exit" is always saved as the most recent command when exiting IHx. For ergonomic use I think the "exit" should not be saved, but didn't want to go too far into specifics without feedback.
    - Since my own use-case is to import ConsoleReader for my own language's repl, it wouldn't make sense to hard-code the string "exit" into ConsoleReader.
- Maybe the default maximum for commands should be something like 50, as opposed to unlimited.
- The history file is written to after each command is entered. Maybe it would be preferable to only write on exit.
- If the history file can't be opened for writing, the user won't get an error until entering their first command--so maybe that exception should be checked for in the constructor instead of at save-time.

ianxm commented on src/ihx/History.hx:
```diff
@@ -22,17 +22,51 @@ package ihx;
+                commands = sys.io.File.getContent(saveFile).split("\n");
+                commands.insert(0, "");
+            }
+            catch ( err: Dynamic ) {
+                commands = [""];
```
this should print a warning. something like "unable to open history file, not loading history; [err]"

ianxm commented on src/ihx/History.hx:
```diff
@@ -22,17 +22,51 @@ package ihx;
+        if ( saveFile.length > 0 )
+        {
+            // Save the command history without the empty string at index 0.
+            sys.io.File.saveContent(saveFile, commands.slice(1).join("\n"));
```
if this fails, it should print a warning and move on.

ianxm commented on src/ihx/History.hx:
```diff
@@ -22,17 +22,51 @@ package ihx;
     public function add(cmd)
     {
         commands.push(cmd);
+        if ( maxCommands > 0 && commands.length > maxCommands ) 
```
`maxCommands > 0 && commands.length > maxCommands` is sort of an alias for `shouldSaveHistory`. you could define a private variable or method to not have to repeat it and also make it clear what you're checking for.

ianxm commented:
I think it makes sense to save 'exit.' bash saves 'exit.'

default max of 50 makes sense.

only writing the history file on exit would be better.

it's fine to not check if the history file isn't checked until the first command, but don't let the error propagate and kill ihx if there is a problem loading or saving the history file. just print a warning.

NQNStudios commented:
I think my new commit addresses your review. To make the history save on exit-only, changes were required in more files than I expected -- in particular, I hope passing a `quit()` function from `IHx` to the `CommandProcessor` constructor isn't too odd. This way, the object composition doesn't need to dramatically change in such a way that the CommandProcessor would know about the ConsoleReader.

NQNStudios commented:
Done and done

ianxm commented:
merged
